File: web/src/api/position.js
import service from '@/utils/service'

export function getPositionList(params) {
  return service({
    url: '/api/position/list',
    method: 'get',
    params
  })
}

export function getDevicePosition(id) {
  return service({
    url: `/api/position/${id}`,
    method: 'get'
  })
}

export function getPositionTrack(id) {
  return service({
    url: `/api/position/${id}/track`,
    method: 'get'
  })
}

export function deletePosition(id) {
  return service({
    url: `/api/position/${id}`,
    method: 'delete'
  })
}

File: server/router/gb28181/position.go
package gb28181

import (
	"wvp-go/server/api/v1/gb28181"

	"github.com/gin-gonic/gin"
)

type PositionRouter struct{}

var positionRouter = new(PositionRouter)

func (p *PositionRouter) InitPositionRouter(Router *gin.RouterGroup) {
	positionRouterGroup := Router.Group("/position")
	{
		positionRouterGroup.GET("/list", gb28181.GetPositionList)
		positionRouterGroup.GET("/:id", gb28181.GetDevicePosition)
		positionRouterGroup.GET("/:id/track", gb28181.GetPositionTrack)
		positionRouterGroup.DELETE("/:id", gb28181.DeletePosition)
	}
}
